### PR TITLE
adding tests for exec packages

### PR DESCRIPTION
**Pull request by mohamedasifs123**
#### What type of PR is this?

adding tests for exec packages.
it is a draft pr. will add tests for other files too.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: pkg/utils/exec/cmd_other_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+// Helper function to convert int64 to *int64
+func intPtr(n int64) *int64 {
+	return &n
+}
+
+type args struct {
+	uid *int64
+	gid *int64
+}
```
Comment (wzshiming) [inline pkg/utils/exec/cmd_other_test.go:230]: Move into TestSetUser

File: pkg/utils/exec/cmd_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		t.Fatalf("failed to kill test process: %v", err)
+	}
+
+	// Wait for the process to exit
+	if _, err := cmd.Process.Wait(); err != nil {
+		t.Fatalf("failed to wait for process to exit: %v", err)
+	}
+
+	// Ensure the process is no longer running
+	time.Sleep(1 * time.Second) // Give the system some time to update the process state
```
Comment (wzshiming) [inline pkg/utils/exec/cmd_test.go:74]: The `cmd.Process.Wait()` above should have secured this
```suggestion
```

File: pkg/utils/exec/cmd_other_test.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := startProcess(tt.args.ctx, tt.args.name, tt.args.arg...); !reflect.DeepEqual(got.Args, tt.want.Args) {
+				t.Errorf("startProcess() = want %v got %v", tt.want.Args, got.Args)
+			}
+		})
+	}
+}
+
+func Test_command(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (mohamedasifs123) [reply pkg/utils/exec/cmd_other_test.go:77]: Okay will remove it

File: pkg/utils/exec/cmd_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		t.Fatalf("failed to kill process %d: %v", pid, err)
+	}
+
+	// Ensure the process is no longer running
+	if IsRunning(pid) {
+		t.Fatalf("expected process %d to be stopped", pid)
+	}
+}
+
+func TestIsRunning(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (mohamedasifs123) [reply pkg/utils/exec/cmd_test.go:49]: Yeh the IsRunning calls the is running.
I'll remove the one .

File: pkg/utils/exec/version_test.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+	tests := []struct {
+		name    string
+		args    args
+		want    version.Version
+		wantErr bool
+	}{{
+		name: "testing go",
+		args: args{
+			ctx:  context.Background(),
+			path: "go",
```
[the remark this answers is not in the document]
Comment (wzshiming) [reply pkg/utils/exec/version_test.go:43]: Not yet.

It's not easy to test for coverage, so let's remove the file and leave it for next time.

File: pkg/utils/exec/cmd_test.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+	err := KillProcess(pid)
+	if err != nil {
+		t.Fatalf("failed to kill process %d: %v", pid, err)
+	}
+	err = cmd.Wait()
+	if err != nil {
+		// Check if the error indicates the process was killed
+		var exiterr *exec.ExitError
+		if errors.As(err, &exiterr) {
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
```
Comment (wzshiming) [inline pkg/utils/exec/cmd_test.go:49]: This won't work on Windows

File: pkg/utils/exec/cmd_test.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+				if status.Signaled() && status.Signal() == syscall.SIGKILL {
+					t.Logf("process %d was killed as expected", pid)
+				} else {
+					t.Fatalf("process %d exited with status %v, expected to be killed", pid, status)
+				}
+			} else {
+				t.Fatalf("failed to get exit status for process %d: %v", pid, err)
+			}
+		}
```
Comment (wzshiming) [inline pkg/utils/exec/cmd_test.go:58]:
```suggestion
		} else {
		   // TODO
		}
```

File: pkg/utils/exec/version_test.go [hunk 8]
```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
Comment (wzshiming) [inline pkg/utils/exec/version_test.go:2]:
```suggestion
Copyright 2024 The Kubernetes Authors.
```

File: pkg/utils/exec/cmd_other_test.go [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+	// Run tests
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := setUser(cmd, tt.args.uid, tt.args.gid)
+			if err != nil {
+				t.Errorf("setUser() error = %v, want nil", err)
+				return
+			}
+		})
```
[the remark this answers is not in the document]
Comment (wzshiming) [reply pkg/utils/exec/cmd_other_test.go:103]: Check the GID and UID

File: pkg/utils/exec/cmd_test.go [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+	if !IsRunning(pid) {
+		t.Fatalf("expected process %d to be running", pid)
+	}
+
+	// Kill the process
+	err := KillProcess(pid)
+	if err != nil {
+		t.Fatalf("failed to kill process %d: %v", pid, err)
+	}
+	err = cmd.Wait()
```
Comment (wzshiming) [inline pkg/utils/exec/cmd_test.go:48]:
``` golang
	err = cmd.Wait()
	if err == nil {
		t.Fatalf("process %d exited normally, expected to be killed", pid)
	}

	// Check if the error indicates the process was killed
	var exiterr *exec.ExitError
	if !errors.As(err, &exiterr) {
		t.Fatalf("process %d, unexpected errror %#v", pid, err)
	}
```

> process 94432, unexpected errror &os.SyscallError{Syscall:"wait", Err:0xa}

**Pull request closed without merging**